crates/evm/src/execute.rs

```diff
@@ -179,6 +179,27 @@ pub trait BlockExecutorProvider: Send + Sync + Clone + Unpin + 'static {
         DB: Database<Error: Into<ProviderError> + Display>;
 }
 
+/// Executor hooks to extend the behaviour of [`Executor`].
+///
+/// The hooks are inserted in [`Executor::execute`] lifecycle and called in the following order:
+/// 1. [`ExecutorHooks::pre_state_changes`]
+/// 1. Executor does pre-block state changes, such as system contract calls.
+/// 1. [`ExecutorHooks::pre_transactions`]
+/// 1. Executor executes transactions
+/// 1. [`ExecutorHooks::post_transactins`]
+/// 1. Executor does post-block state changes, such as system contract calls.
+/// 1. [`ExecutorHooks::post_state_changes`]
+pub trait ExecutorHooks {
```

Comment:
I don't really like the name hook because these are more like handlers that take care of transacting a block